script/bulk_import/generic_bulk.rb

```diff
@@ -623,10 +623,14 @@ def import_user_associated_accounts
     SQL
 
     existing_user_ids = UserAssociatedAccount.pluck(:user_id).to_set
+    existing_provider_uids = UserAssociatedAccount.pluck(:provider_uid).to_set
 
     create_user_associated_accounts(accounts) do |row|
       user_id = user_id_from_imported_id(row["user_id"])
+      provider_uid = row["provider_uid"]
+
       next if user_id && existing_user_ids.include?(user_id)
+      next if provider_uid && existing_provider_uids.include?(provider_uid)
```

Comment:
Good idea! Let's check it with `provider_name` though because there's the slight possibility that the same `provider_uid` is used with different providers.

```suggestion
    existing_provider_uids = UserAssociatedAccount.pluck(:provider_uid, :provider_name).to_set

    create_user_associated_accounts(accounts) do |row|
      user_id = user_id_from_imported_id(row["user_id"])

      next if user_id && existing_user_ids.include?(user_id)
      next if provider_uid && existing_provider_uids.include?([row["provider_uid"], row["provider_name"]])
```

```suggestion
    existing_provider_uids = UserAssociatedAccount.pluck(:provider_uid, :provider_name).to_set
```